### PR TITLE
Tnw tool

### PR DESCRIPTION
**Pull request by jaxonwillard**
This integrates the TNW test tool page and refactors tnw.js. I also pulled out uttam's DTW tool completely

### CODE REVIEW

File: TNW.html [hunk 1]
```diff
@@ -44,14 +44,7 @@
 <body>
 <div id="root">
     <header class="header">
-        <button class="match-a-to-b">MATCH A TO B</button>
```
Comment (jaxonwillard) [inline TNW.html:47]: I pulled all these out to keep it simple

File: js/TNWTool.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
+const root = d3.select('#root');
```
Comment (jaxonwillard) [inline js/TNWTool.js:1]: This file has all the actions needed to run TNW.html

File: js/reorder/distanceFunctions/tnw.js [hunk 3]
```diff
@@ -4,22 +4,19 @@ function calculateTNWDistance(sim1, sim2) {
     return tndDist * -1;
 }
 
-function getTNWScore(sim1, sim2) {
-    // console.log('Calculating TNW score ', simulation1, simulation2);
-    let simulation1 = sim1.events.filter((d) => { return d.on; });
-    let simulation2 = sim2.events.filter((d) => { return d.on; });
+function getTNWScore(sim1, sim2, GAP_SCORE=-1, MATCH_SCORE=1, MAX_OFFSET_PENALTY=-1, dataFilter = (simulation) => simulation.events.filter((d) => { return d.on; }), dataSelector = d => d.t, resultSelector = (matrix) => matrix[matrix.length - 1][(matrix[matrix.length - 1].length - 1)]) {
```
Comment (jaxonwillard) [inline js/reorder/distanceFunctions/tnw.js:7]: This generalizes the function so you can choose what the data looks like. By default, it takes the same parameters as before.

File: js/reorder/distanceFunctions/tnw.js [hunk 4]
```diff
@@ -36,7 +33,7 @@ function getTNWScore(sim1, sim2) {
     for (let i = 1; i < simulation1.length + 1; i++) {
         for (let j = 1; j < simulation2.length + 1; j++) {
             let arrowImage = 'd.png'
-            let match = matrix[i - 1][j - 1].cellMax + MATCH_SCORE + offsetPenalty(simulation1[i], simulation2[j], MAX_OFFSET_PENALTY)
```
Comment (jaxonwillard) [inline js/reorder/distanceFunctions/tnw.js:39]: This was a bug that got fixed in earlier branches. This doesn't solve the "temporal" part of the algo that we talked about on 9/3

File: js/reorder/distanceFunctions/tnw.js [hunk 5]
```diff
@@ -66,36 +63,17 @@ function getTNWScore(sim1, sim2) {
...
 
-    event1 = event1.t
-    event2 = event2.t
+    event1 = dataSelector(event1)
+    event2 = dataSelector(event2)
 
     return MAX_OFFSET_PENALTY * (Math.abs(event1 - event2) / Math.max(event1, event2))
 }
 
-function makeTable(data) {
```
Comment (jaxonwillard) [inline js/reorder/distanceFunctions/tnw.js:86]: I put this into TNWTool.js

**Review by stevescott32 — APPROVED** [review]
Everything I tried out worked great, looks good.

**Pull request merged**